### PR TITLE
chore: switch to ESM

### PR DESCRIPTION
**Pull request by king-11**
**What is the purpose of this pull request? (put an "X" next to item)**

- [x] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Other, please explain:

**What changes did you make? (Give an overview)**

- refactor: update import/export of source files
- refactor: update common test file
- chore: update import in tests
- chore: update package.json for esm support
- chore: update Readme esm

**Which issue (if any) does this pull request address?**

- https://github.com/webtorrent/webtorrent/issues/2177

**Is there anything you'd like reviewers to focus on?**

### CODE REVIEW

**Comment by king-11** [comment]
@feross @DiegoRBaquero any reviews on this one

**Comment by king-11** [comment]
@DiegoRBaquero any update on this?

**Comment by ThaUnknown** [comment]
it's likely he doesnt have the time as of now, but might soon

If i was you i'd do these in batches of like 10 at a time tbh x)

**Comment by king-11** [comment]
@ThaUnknown 😂 i did rather wait...can work on PRs for other repository by the time this one gets reviewed

**Comment by ThaUnknown** [comment]
> @ThaUnknown 😂 i did rather wait...can work on PRs for other repository by the time this one gets reviewed

you definitely should, then the person reviewing these PR's can just sit down for 1-2h and do it all in a batch, rather than one at a time

File: README.md [hunk 1]
```diff
@@ -41,8 +41,8 @@ npm install magnet-uri
 ```
 
 ```js
-const DHT = require('bittorrent-dht')
-const magnet = require('magnet-uri')
+import DHT from 'bittorrent-dht'
+import magner from 'magnet-uri'
```
Comment (DiegoRBaquero) [inline README.md:45]:
```suggestion
import magnet from 'magnet-uri'
```

File: package.json [hunk 2]
```diff
@@ -42,10 +43,12 @@
     "peer-to-peer"
   ],
   "engines": {
-    "node": ">=10"
+    "node": ">=12.10.0"
```
Comment (DiegoRBaquero) [inline package.json:46]:
```suggestion
    "node": ">=12.20.0"
```

File: test/dht_store_mutable.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
-const common = require('./common')
-const DHT = require('../')
-const ed = require('bittorrent-dht-sodium')
-const test = require('tape')
-const crypto = require('crypto')
+import ed from 'bittorrent-dht-sodium'
+import crypto from 'crypto'
```
Comment (DiegoRBaquero) [inline test/dht_store_mutable.js:2]:
```suggestion
import crypto from 'node:crypto'
```

Also make it first?

File: test/dht_test_vectors.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
-const common = require('./common')
-const DHT = require('../')
-const test = require('tape')
-const crypto = require('crypto')
-const ed = require('bittorrent-dht-sodium')
+import ed from 'bittorrent-dht-sodium'
+import crypto from 'crypto'
```
Comment (DiegoRBaquero) [inline test/dht_test_vectors.js:2]: same

File: test/dht_store_mutable.js (hunk 3 above)
Comment (king-11) [reply test/dht_store_mutable.js]: imports are sorted by module name for now shall we change that?

Comment (DiegoRBaquero) [reply test/dht_store_mutable.js]: Yes please!

**Review by ThaUnknown — APPROVED** [review]

**Pull request merged**